### PR TITLE
Fix compilation error

### LINKED ISSUE #48: Build fails on mac
```
master~/.vim/plugged/cpsm$ ./install.sh
PY3 not specified; inferring Python version from /usr/local/bin/vim
Python 3 selected
-- The C compiler identification is AppleClang 9.1.0.9020039
-- The CXX compiler identification is AppleClang 9.1.0.9020039
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - found
-- Found Threads: TRUE
-- Boost version: 1.67.0
-- Found the following Boost libraries:
--   program_options
-- Found PythonInterp: /usr/local/bin/python3 (found suitable version "3.7", minimum required is "3")
-- Found PythonInterp: /usr/local/bin/python3 (found version "3.7")
-- Found PythonConfig: -I/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m -I/usr/local/Cellar/python/3.7.0/Frameworks/Python.framework/Versions/3.7/include/python3.7m; -L/usr/local/opt/python/Frameworks/Python.framework/Versions/3.7/lib/python3.7/config-3.7m-darwin -lpython3.7m -ldl -framework CoreFoundation
-- Optional package ICU was not found
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/xxx/.vim/plugged/cpsm/build
Scanning dependencies of target cpsm_core
[ 11%] Building CXX object CMakeFiles/cpsm_core.dir/src/str_util.cc.o
[ 22%] Linking CXX static library libcpsm_core.a
[ 22%] Built target cpsm_core
Scanning dependencies of target cpsm_cli
[ 33%] Building CXX object CMakeFiles/cpsm_cli.dir/src/cpsm_cli_main.cc.o
[ 44%] Linking CXX executable cpsm_cli
[ 44%] Built target cpsm_cli
Scanning dependencies of target cpsm_py
[ 55%] Building CXX object CMakeFiles/cpsm_py.dir/src/ctrlp_util.cc.o
[ 66%] Building CXX object CMakeFiles/cpsm_py.dir/src/python_extension.cc.o
/Users/xxx/.vim/plugged/cpsm/src/python_extension.cc:48:11: error: assigning to 'char *' from incompatible type 'const char *'
  *data = PyUnicode_AsUTF8AndSize(obj, size);
          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/cpsm_py.dir/src/python_extension.cc.o] Error 1
make[1]: *** [CMakeFiles/cpsm_py.dir/all] Error 2
```

### PR DESCRIPTION
**Pull request by ptzz**
Fixes #48.

### CODE REVIEW

**Comment by maxnordlund** [comment]
I also encountered this today, and unfortunately the Travis build fails on python 2. It seems like they may have changed the type with 3.7

**Comment by ptzz** [comment]
Ah, didn’t check CI results 😕. I’m without computer access for a couple of weeks, anyone feel free to push fixes to this PR meanwhile.

**Comment by ptzz** [comment]
Turns out I forgot to remove the initial hacky cast solution in the PR. But keeping that (line 48) and removing the “nice” changes probably makes CI happy. Maybe the cast is acceptable in this case anyway.

**Comment by maxnordlund** [comment]
Can I push to your branch? Or could you give me access to your fork; I could take a stab at it.

**Comment by ptzz** [comment]
@maxnordlund Sure, did you try? Not sure how it works with permissions, some guy pushed to a PR of mine the other day without me giving explicit permissions. Pushed to my fork that is.

**Comment by maxnordlund** [comment]
I think that only works for members of the original repo, i.e. nixprime.

**Comment by ptzz** [comment]
@maxnordlund I’ve sent you a collaborator invite.

**Comment by maxnordlund** [comment]
Thank you, I've accepted and will try some stuff.

**Comment by maxnordlund** [comment]
Well, that went smoothly. Now it's ready I think, if @nixprime agrees. (Also you can just squash away my commit)

**Comment by ptzz** [comment]
@maxnordlund 🎉💪👍

File: src/python_extension.cc [hunk 1]
```diff
@@ -45,7 +45,7 @@ typedef std::unique_ptr<PyObject, PyObjectDeleter> PyObjPtr;
 inline bool PyVimString_AsStringAndSize(PyObject* obj, char** data,
                                         Py_ssize_t* size) {
 #if PY_MAJOR_VERSION >= 3
-  *data = PyUnicode_AsUTF8AndSize(obj, size);
+  *data = (char*)PyUnicode_AsUTF8AndSize(obj, size);
```
Comment (jparise) [inline src/python_extension.cc]: `const_cast<char *>(PyUnicode_AsUTF8AndSize(obj, size))` would be more idiomatic for C++.

Comment (maxnordlund) [reply src/python_extension.cc]: OK, I'll change it.

**Comment by Shougo** [comment]
@nixprime Ping.  I have encountered the problem today..

**Comment by danarnold** [comment]
This PR fixes the issue for me with nvim and python 3.7.0-3. Thanks @ptzz and @maxnordlund !

**Pull request merged**